src/srn.ts

```diff
@@ -0,0 +1,33 @@
+import { extname } from './extname';
+
+export interface IDeserializedSrn {
+  shortcode: string;
+  orgSlug: string;
+  projectSlug: string;
+  uri?: string;
+  file?: string;
+  ext?: string;
+}
+
+export function deserializeSrn(srn: string): IDeserializedSrn {
+  const [shortcode, orgSlug, projectSlug, ...uriParts] = srn.split('/');
+
+  const uri = uriParts.length ? `/${uriParts.join('/')}` : ``;
+  const file = uri
+    .split(/(.json|.yaml|.yml|.md)/)
```

Comment:
I suspect you want to match extension here or something, so the dot here won't work properly.
Dot in regular expressions matches anything but newline, so
```suggestion
    .split(/(\.(?:json|ya?ml|md))/)
```

Reply:
Yeah, it does do anything.

Reply:
@wmhilton I actually have to admit that Tom inspired me.
Now, whenever y'all yell at me because I use regular expressions and urge me to use 10932819821921-long-method-chain of split/join/whatever, I'll just use split and put regular expression :laughing: